### PR TITLE
New103

### PR DESCRIPTION
**Pull request by shawn320**
# Description

- call API to fetch contacts and populate dropdown and display contact & enagement info of selected contact
- updated with main branch
- added some stylings on the engagement form
 
Note: This is the updated version of my previous pr. Fixed some bugs that Beniot pointed out earlier.

## Test Instructions

- navigate to the add contacts & engagement page, select engagement
- play with the dropdown to see if contacts displayed properly
- engagement info should not be displayed if the contact selected doen't have any engagements 

## Reviewers
@bcloutier7 @P0NDER0SA @will0684 @andr0272 @MarcoGoC @nghe0001 @shewood 

## Checklist
- [x] Strings use placeholders for translation (No hard coded strings)
- [ ] Unit tests have been added/updated

### CODE REVIEW

File: components/engagement/EngSelectContacts.vue [hunk 1]
```diff
@@ -133,13 +173,23 @@ td {
...
 button {
   @apply font-bold text-lg
 }
-
 button:focus,
 option:focus {
   @apply outline-none;
 }
 </style>
+© 2020 GitHub, Inc.
```
Comment (P0NDER0SA) [inline components/engagement/EngSelectContacts.vue:184]: I noticed these Github lines in a couple of files.  might want to remove those there Bud...  besides that it tests well

Comment (shawn320) [reply components/engagement/EngSelectContacts.vue]: Thanks for point it out, it's fixed

File: components/engagement/EngFormSelect.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+.formselect:focus {
+  @apply outline-none border-blue-500;
+}
+</style>
+© 2020 GitHub, Inc.
+Terms
+Privacy
+Security
+Status
```
Comment (P0NDER0SA) [inline components/engagement/EngFormSelect.vue:46]: same here :)

File: components/app/AppDisplayContact.vue [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<template>
+  <div class="show-contact grid grid-cols-2 gap-4">
+    <div class="border-t border-b border-gray-400 p-5">
+      <span>
+        {{ name }}
+      </span>, {{ orgName }}
+      <br />
+      {{ email }}
+    </div>
+    <div v-if="type != ''" class="border-t border-b border-gray-400 p-5">
```
Comment (MarcoGoC) [inline components/app/AppDisplayContact.vue:10]: I believe in vue v-if="type !=''" is the same as just v-if="type"

Comment (MarcoGoC) [reply components/app/AppDisplayContact.vue:10]: Also why are you using a new component when there is already one ?

File: components/engagement/EngSelectContacts.vue [hunk 4]
```diff
@@ -41,65 +44,106 @@
...
       </button>
     </div>
+    -->
   </div>
 </template>
 
 <script>
-import AppSelect from '../app/AppSelect'
-import showContacts from './EngShowContacts'
+import displayContact from '../app/AppDisplayContact'
```
Comment (MarcoGoC) [inline components/engagement/EngSelectContacts.vue:76]: you should be using the component from task # 111

File: components/engagement/EngSelectContacts.vue [hunk 5]
```diff
@@ -41,65 +44,106 @@
...
       moreContacts: false,
-      isSelected1: false,
-      isSelected2: false
+      isSelected: false
+    }
+  },
+  // eslint-disable-next-line space-before-function-paren
+  async created() {
+    try {
+      const res = await this.$axios.get('/api/contact/contacts')
```
Comment (MarcoGoC) [inline components/engagement/EngSelectContacts.vue:104]: use the Vuex store call for this call

Comment (shawn320) [reply components/engagement/EngSelectContacts.vue:104]: I'll do this on a new branch, and do a pr after

File: components/engagement/EngSelectContacts.vue (hunk 4 above)
Comment (shawn320) [reply components/engagement/EngSelectContacts.vue]: This is fixed, I am using the component you created

File: components/app/AppDisplayContact.vue (hunk 3 above)
Comment (shawn320) [reply components/app/AppDisplayContact.vue]: ok, it works

File: components/engagement/EngSelectContacts.vue (hunk 5 above)
Comment (P0NDER0SA) [reply components/engagement/EngSelectContacts.vue:104]: the boys are gonna set up a separate PR for that

**Pull request merged**